### ISSUE TITLE
ContentChildren not working by Component ref.

### ISSUE THREAD

**Issue by damiandennis** [angular/angular#11645]
**I'm submitting a ...**  (check one with "x")
```
[x ] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior** 
Can no longer fetch content children that are not direct children of the parent component.

**Expected behavior**
I should be able to fetch the children no matter where they are nested in the parent component.

**Reproduction of the problem**
http://plnkr.co/edit/F4DwHzL1I8ByT6y6slsk

**What is the motivation / use case for changing the behavior?**
It has worked all through beta/rc releases but now not on go live.

**Please tell us about your environment:**
Chrome, Ubuntu

* **Angular version:** 
2.0.0

* **Browser:** 
all
 
* **Language:**
all
 
* **Node (for AoT issues):** `node --version` =

**Comment by vicb** [comment]
what are the steps to reproduce with your plunker ?

**Comment by damiandennis** [comment]
Sorry I have not done the example right, need to correct to ContentChild, using it like ViewChild in my plunker but the bug is still there just not showing it correctly.

**Comment by damiandennis** [comment]
ok I have corrected the plunker to my scenario, it seems the problem is that its nested under a element with a directive, this used to work but no longer does.
http://plnkr.co/edit/F4DwHzL1I8ByT6y6slsk

**Comment by damiandennis** [comment]
Steps:

1. Run the plunker
2. You should get a alert saying that the component was not found.
3. Look at the code in component1 and the view of the app and see that it can not find the component2 because there is an element above it that has a directive. component3 however sits outside of the element with the directive and can be found.

**Comment by vicb** [comment]
Up until when was this working ?
Please provide a plunker for the last working release. Thanks.

**Comment by damiandennis** [comment]
It was working in rc.6 I did not get a chance to update to rc.7

**Comment by damiandennis** [comment]
having a bit of trouble setting up the rc.6 plunker (permission denied), got a link to one that I can use as a starting point?

**Comment by zoechi** [comment]
@damiandennis you can just use the `new` button in the Plunker editor. There is an Angular2 TS template or the live example link in https://angular.io/docs/ts/latest/quickstart.html or when you create a new issue you also get a link to a Angular2 template.

**Comment by damiandennis** [comment]
There is no rc6 template, actually have tried to find one, also trying to change unpkg.com to use rc6 version but does not work says invalid url.
https://unpkg.com/@angular@2.0.0-rc.6/core/bundles/core.umd.js

**Comment by damiandennis** [comment]
actually figured out what I was doing wrong should have been https://unpkg.com/@angular/core@2.0.0-rc.6/bundles/core.umd.js

**Comment by damiandennis** [comment]
He is it working on rc6 http://plnkr.co/edit/tAPFuBbsAKMAdI2SdqiH?p=preview

**Comment by damiandennis** [comment]
as you can see there is no alert for the component2 like there is in the 2.0.0 version

**Comment by damiandennis** [comment]
its is broken as well however in rc7 http://plnkr.co/edit/jkWBJzsdDmBU3vyeVuEP?p=preview

**Comment by zoechi** [comment]
Seems you need `{descendants: true}` http://plnkr.co/edit/3yYvwri1HkThE3uR7U3c?p=preview

GitHub issues are for bug reports and feature requests.
    For support questions please use other channels like the ones listed in [CONTRIBUTING - Got a Question or Problem?](https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question)

**Comment by damiandennis** [comment]
Thanks, that does solve it, this is a breaking change that should be in the CHANGELOG, I can see no reference to it. It is fair enough that I reported this as a bug because as far as I could tell it was.

**Comment by vicb** [comment]
@damiandennis @zoechi this is a real issue in our code:
1- you should not need explicit `{descendants: true}` - this should be the default,
2- you probably should not be able to override this default (at least type checking should error).

Will fix this in a coming PR.
Thanks for the report and the repros !

**Comment by vicb** [comment]
fixed in #11672
